README.tmpl.md

```diff
@@ -0,0 +1,206 @@
+found in `man (5) hunspell`.
+
+### `.dic` Files
+
+The `.dic` file comprises what you would normally think of as a dictionary.
+It is a list of words, one word per line, with an initial hint to hunspell
+indicating how many words it should expect to work with. The value of the
+hint is not particularly important except for performance, but must be
+greater than zero.
+
+In addition to the words, hunspell allows the dictionary to encode "affix rules".
+These describe how word roots can be extended to allow related words to be
+matched as correct, for example "thing" and "things", or "lint" and "linting".
+The affix rules are indicated by a set of characters following the word and
+separated by a slash. The two examples here would be represented (in the
+en_US case) by
+```
+lint/G
+thing/S
+```
+where the `/S` indicats that "thing" can be pluralised and the "lint" can be
+extended to its gerund. It is possible to specify more than one affix rule,
+and affixes can be prefix or suffix modifiers. Prefix and suffix rules can
+interact as the cross product (see below). So from the en_US dictionary,
+```
+advise/LDRSZGB
+advised/UY
+```
+will match "advise" (root), "advisement" (L), "advised" (D), "adviser" (R),
+"advises" (S), "advisers" (Z), "advising" (G) and "advisable" (B) from the
+first rule, and "advised" (root), "unadvised" (U), "advisedly" (Y) and
+"unadvisedly" (UY) from the second.
+
+### `.aff` Files
+
+Different natural languages use different inflection constructions for
+encoding grammatical information and this is specified for each language
+in hunspell's `.aff` files.
+
+Again using the en_US language locale, an example of an affix rule definition
+(the Z from "advise" above)
+```
+SFX Z Y 4
+SFX Z   0     rs         e
+SFX Z   y     iers       [^aeiou]y
+SFX Z   0     ers        [aeiou]y
+SFX Z   0     ers        [^ey]
+```
+The first line indicated that the rule is the application of a suffix
+(SFX), the rule name is "Z", that the rule can be combined with prefixes
+via cross product (Y) and that there are 4 ways to apply the rule.
+
+The remaining lines indicate how each application of the rule should be
+applied, and when. The first last field specifies when each rule can be
+applied matching to the target word. The second last indicates the suffix to
+add, and the third last indicates any characters to remove before adding the
+suffix ("0" indicated no removal). So with "advise", the first rule matches,
+resulting in "advise"+"r" being accepted as a correctly spelling word. The
+words "buy" and "fly" illustrate how the second and third rules would by
+applied; "buy" will match the third rule and so would allow "buy"+"ers",
+but "fly" would match the second and allow "fl"+"iers".
+
+If you are adding specific rules to your `.words` file, the `.aff` files
+for your system can be found in the hunspell dictionary path for reference.
+Invoking `hunspell -D` will print the search path and show which dictionaries
+are available.
```

Comment:
It would be nice to give an example of where the en_US affix file will generally be located, as that will make it easier for most new users

Reply:
Will add.